subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java

```diff
@@ -446,15 +446,17 @@ public String determineName(Configuration c) {
     ResolvableDependencies getIncoming();
 
     /**
-     * Returns the outgoing artifacts of this configuration.
+     * Returns the outgoing publications object containing artifacts and variants published by this configuration.
```

Comment:
I worry about overloading the term `publications` here since the data provided by this object is not necessarily always _published_, in a maven/ivy sense, to an actual repository. Granted, I understand the type is called `ConfigurationPublications`, though IMO that is also kinda misnamed since while this data is used during publication, it is also used during local dependency resolution. 

I guess we need to nail down how we want to define publishing here. Is it okay that we're overloading publishing as both a concept of a component advertising its variants to dependency management and also the concept of a component getting serialized in some manner and uploaded to a remote repository?

Reply:
`ConfigurationPublications` I agree is poorly named.  

I think it is inaccurate to say this returns "artifacts" and that "publications object" is a marginal improvement - especially with the context after the name - but it's not great.  Can you think of something else we could say that would be clearer?  I think the problem is that this type doesn't really map to anything in the conceptual domain of publishing, it's more of an implementation detail.

> Is it okay that we're overloading publishing as both a concept of a component advertising its variants to dependency management and also the concept of a component getting serialized in some manner and uploaded to a remote repository?

I think this is an excellent point and we should discuss it further.

Reply:
Looking at this again, I agree just saying `artifacts` like it was before is inaccurate and `publications object` is better. You could also consider just `@link`ing the type. Otherwise I have no other recommendations right now, but I am happy to discuss further. For example, we could write any of the following, but I am fine with whats here now:
`Returns the outgoing {@link ConfigurationPublications} containing....`
`Returns the {@link ConfigurationPublications} object containing...`
`Returns the outgoing {@link ConfigurationPublications} object containing...`

In general, I have been using the term `advertising` for this kind of behavior. For example, a component "advertises" its variants to dependency-management. I'm not super tied to this term so I'm definitely open to other ideas, though I like this better than saying publishing in reference to local resolution.

Reply:
I like the word advertising for this, here and in general, and linking is a good idea.  Will update.